### PR TITLE
chore(docs): migrate dev-novel to storybook (steps 1-2)

### LINKED ISSUE #226: Move from DevNovel to Storybook
Move our showcase from [DevNovel](https://github.com/algolia/dev-novel) to [Storybook](https://storybook.js.org/basics/guide-angular/) to leverage the community & plugins of Storybook. It also brings consistency across our different flavours since React & Vue are already on Storybook. 

Related https://github.com/algolia/instantsearch.js/issues/3051

### PR DESCRIPTION
**Pull request by tkrugg**
DEMO: https://deploy-preview-368--angular-instantsearch.netlify.com/examples/storybook/?selectedKind=InstantSearchJS&selectedStory=with%20algoliasearch%20search%20client&full=0&addons=1&stories=1&panelRight=0

Purpose: migrate dev-novel to storybook ref #226 

What this PR does:
- set up initial boilerplate for storybook 
- set build for storybook

Action plan: https://github.com/algolia/angular-instantsearch/issues/226#issuecomment-435846001

### CODE REVIEW

**Comment by Haroenv** [comment]
seems like there's some linting mistake that broke circle

File: examples/storybook/src/stories/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    imports: [NgAisModule.forRoot()],
+    declarations: [WithSearchClient],
+    providers: [],
+  })
+);
+stories
+  .add('with algoliasearch search client', () => ({
+    component: WithSearchClient
+  }));
+
```
Comment (tkrugg) [inline examples/storybook/src/stories/index.ts]: The list of examples is minimale for now
screenshot at nov 05 11-16-20

File: examples/storybook/src/wrap-with-hits.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import { Component } from '@angular/core';
+
+type Helper = {
+  search: Function;
+  addDisjunctiveFacetRefinement: Function;
+};
+
+export function wrapWithHits({
```
Comment (tkrugg) [inline examples/storybook/src/wrap-with-hits.ts]: This helper function was copied almost as is from dev-novel 
https://github.com/algolia/angular-instantsearch/blob/develop/examples/dev-novel/wrap-with-hits.ts

File: examples/storybook/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "@angular/forms": "7.0.0",
+    "@angular/http": "7.0.0",
+    "@angular/platform-browser": "7.0.0",
+    "@angular/platform-browser-dynamic": "7.0.0",
+    "@angular/router": "7.0.0",
+    "angular-instantsearch": "2.1.0",
+    "core-js": "2.5.4",
+    "rxjs": "6.3.3",
+    "zone.js": "0.8.26"
+  },
```
Comment (tkrugg) [inline examples/storybook/package.json]: I would remove some of theses dependencies but I'm not sure yet storybook isn't using them.
I suspect some of them are silently used by storybook (I broke it by removing @angular/forms)
Once I've finished loading all stories I'll remove them

Comment (tkrugg) [reply examples/storybook/package.json]: done in #269

**Pull request merged**